src/processors/physics.rs

```diff
@@ -0,0 +1,127 @@
+//! Physics processor based on `nphysics3d`
+
+extern crate ncollide;
+extern crate nphysics3d;
+
+use std::ops::{Deref, DerefMut};
+
+use components::physics::PhysicsComponent;
+use components::transform::LocalTransform;
+use world_resources::Time;
+
+use ecs::{Join, Processor, RunArg};
+use self::nphysics3d::world::World;
+
+
+/// Wrapper around `nphysics`'s `World` that impls `Send`/`Sync`, so that
+/// it can be processed by the ECS system. We take care of ensuring that
+/// there are no data races inside the ECS system, but the compiler can't
+/// infer this, so we have to manually tell it that it's ok.
+pub struct PhysicsWorld(World<f32>);
+
+impl PhysicsWorld {
+    pub fn new() -> PhysicsWorld {
+        PhysicsWorld(World::new())
+    }
+}
+
+unsafe impl Sync for PhysicsWorld { }
+unsafe impl Send for PhysicsWorld { }
+
+impl Deref for PhysicsWorld {
+    type Target = World<f32>;
+    fn deref(&self) -> &World<f32> {
+        &self.0
+    }
+}
+
+impl DerefMut for PhysicsWorld {
+    fn deref_mut(&mut self) -> &mut World<f32> {
+        &mut self.0
+    }
+}
+
+
+/// Handles physics processing for ECS
+pub struct PhysicsProcessor;
+
+impl PhysicsProcessor {
+    pub fn new() -> PhysicsProcessor {
+        PhysicsProcessor { }
+    }
+}
+
+impl Processor<()> for PhysicsProcessor {
+    fn run(&mut self, arg: RunArg, _: ()) {
+        let (
+            mut world,
+            mut handles,
+            mut locals,
+            time,
+        ) = arg.fetch(|w| (
+            w.write_resource::<PhysicsWorld>(),
+            w.write::<PhysicsComponent>(),
```

Comment:
Mostly just because it otherwise gives me a compiler error :P

```
error: no method named `iter` found for type `(&mut ecs::Storage<components::physics::PhysicsComponent, std::sync::RwLockReadGuard<'_, ecs::Allocator>, std::sync::RwLockReadGuard<'_, ecs::MaskedStorage<components::physics::PhysicsComponent>>>, &mut ecs::Storage<components::transform::local_transform::LocalTransform, std::sync::RwLockReadGuard<'_, ecs::Allocator>, std::sync::RwLockWriteGuard<'_, ecs::MaskedStorage<components::transform::local_transform::LocalTransform>>>)` in the current scope
  --> src/processors/physics.rs:73:60
   |
73 |         for (handle, local) in (&mut handles, &mut locals).iter() {
   |                                                            ^^^^
   |
   = note: the method `iter` exists but the following trait bounds were not satisfied: `&mut ecs::Storage<components::physics::PhysicsComponent, std::sync::RwLockReadGuard<'_, ecs::Allocator>, std::sync::RwLockReadGuard<'_, ecs::MaskedStorage<components::physics::PhysicsComponent>>> : ecs::Join`, `std::sync::RwLockReadGuard<'_, ecs::MaskedStorage<components::physics::PhysicsComponent>> : std::ops::DerefMut`
```

Also, the `RigidBody`s (which `PhysicsComponent`s are a wrapper around) are modified in the call to `world.step`, and I believe we need to use `w.write` to make sure that there are no data races while `world.step` is processing.